src/keyboard.c

```diff
@@ -10714,47 +10714,6 @@ The elements of this list correspond to the arguments of
   return list4 (interrupt, flow, meta, quit);
 }
 
-DEFUN ("posn-at-x-y", Fposn_at_x_y, Sposn_at_x_y, 2, 4, 0,
-       doc: /* Return position information for pixel coordinates X and Y.
-By default, X and Y are relative to text area of the selected window.
-Optional third arg FRAME-OR-WINDOW non-nil specifies frame or window.
-If optional fourth arg WHOLE is non-nil, X is relative to the left
-edge of the window.
-
-The return value is similar to a mouse click position:
-   (WINDOW AREA-OR-POS (X . Y) TIMESTAMP OBJECT POS (COL . ROW)
-    IMAGE (DX . DY) (WIDTH . HEIGHT))
-The `posn-' functions access elements of such lists.  */)
-  (Lisp_Object x, Lisp_Object y, Lisp_Object frame_or_window, Lisp_Object whole)
-{
-  CHECK_NUMBER (x);
-  /* We allow X of -1, for the newline in a R2L line that overflowed
-     into the left fringe.  */
-  if (XINT (x) != -1)
-    CHECK_NATNUM (x);
-  CHECK_NATNUM (y);
```

Comment:
This errors if x is -2, and I don't think that's true of the Rust code.

Reply:
Good catch.

Reply:
I am curious how my solution will be received. I also considered and experimented with extending `LispNumber` but that  means constantly verifying the value is not a float. Which is no better.